src/pages/sa-trx/[functionId]/forms/SaleTransactionForm.js

```diff
@@ -0,0 +1,1936 @@
+
+    if (formik.values.items[index].mdType === 2) {
+      if (currentMdAmount < 0 || currentMdAmount > 100) currentMdAmount = 0
+
+      formik.setFieldValue(`items[${index}].mdAmountPct`, 1)
+      formik.setFieldValue(`items[${index}].mdType`, 1)
+      currentMdType = 1
+      formik.setFieldValue(`items[${index}].mdAmount`, parseFloat(currentMdAmount).toFixed(2))
+    } else {
+      if (currentMdAmount < 0 || currentMdAmount > maxClientAmountDiscount) currentMdAmount = 0
+
+      formik.setFieldValue(`items[${index}].mdAmountPct`, 2)
+      formik.setFieldValue(`items[${index}].mdType`, 2)
+      currentMdType = 2
+      formik.setFieldValue(`items[${index}].mdAmount`, parseFloat(currentMdAmount).toFixed(2))
+    }
+
+    const newRow = {
+      ...formik.values.items[index],
+      mdAmount: currentMdAmount,
+      mdType: currentMdType
+    }
+
+    getItemPriceRow(updateRow, newRow, DIRTYFIELD_MDTYPE, true)
+    checkMdAmountPct(newRow, updateRow)
+  }
+
+  async function onWorkFlowClick() {
+    stack({
+      Component: WorkFlow,
+      props: {
+        functionId: functionId,
+        recordId: formik.values.header.recordId
+      },
+      width: 950,
+      height: 600,
+      title: labels.workflow
+    })
+  }
+
+  const onPost = async () => {
+    await postRequest({
+      extension: SaleRepository.SaleTransaction.post,
+      record: JSON.stringify(formik.values.header)
+    })
+
+    toast.success(platformLabels.Posted)
+    invalidate()
+    window.close()
+  }
+
+  const onUnpost = async () => {
+    const res = await postRequest({
+      extension: SaleRepository.SaleTransaction.unpost,
+      record: JSON.stringify(formik.values.header)
+    })
+
+    await refetchForm(res.recordId)
+    toast.success(platformLabels.Posted)
+    invalidate()
+  }
+
+  function openUnpostConfirmation(obj) {
+    stack({
+      Component: StrictUnpostConfirmation,
+      props: {
+        action() {
+          onUnpost(obj)
+        }
+      },
+      width: 500,
+      height: 300,
+      expandable: false,
+      title: platformLabels.UnpostConfirmation
+    })
+  }
+
+  const actions = [
+    {
+      key: 'RecordRemarks',
+      condition: true,
+      onClick: 'onRecordRemarks',
+      disabled: !editMode
+    },
+    {
+      key: 'WorkFlow',
+      condition: true,
+      onClick: onWorkFlowClick,
+      disabled: !editMode
+    },
+    {
+      key: 'FI Trx',
+      condition: true,
+      onClick: 'onClickIT',
+      disabled: !editMode
+    },
+    {
+      key: 'SA Trx',
+      condition: true,
+      onClick: 'onClickSATRX',
+      disabled: !editMode
+    },
+    {
+      key: 'Aging',
+      condition: true,
+      onClick: 'onClickAging',
+      disabled: !editMode
+    },
+    {
+      key: 'GL',
+      condition: true,
+      onClick: 'onClickGL',
+      valuesPath: formik.values.header,
+      disabled: !editMode
+    },
+    {
+      key: 'Post',
```

Comment:
do you need here the post unpost or locked unlocked?

Reply:
I will change it to lock-unlock better